src/ngraph/runtime/gpu/pass/gpu_layout.cpp

```diff
@@ -54,6 +55,33 @@ namespace ngraph
                         rep_slice->set_op_annotations(op_annotations);
                     }
                 }
+                template <>
+                void GPULayout::LAYOUT_DECL(ngraph::op::Reshape)
+                {
+                    auto reshape = static_cast<ngraph::op::Reshape*>(node.get());
+                    if (reshape->get_is_transpose() &&
+                        reshape->get_output_shape().size() ==
+                            reshape->get_argument(0)->get_shape().size())
```

Comment:
does `reshape->get_output_shape().size() ==
                             reshape->get_argument(0)->get_shape().size()` required. Looks fine if we go 1^m to 1^n, or {2, 2} to {4} etc.

Reply:
Actually we don't need to check if it's identical.

Reply:
take it back, here you are check the case that can not do inplace.